ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml

```diff
@@ -503,6 +526,87 @@ SQUtils.QObject {
             }
             return tx
         }
+
+        // returns {
+        //   preparedData,
+        //   value // null or ETH Big number
+        // }
+        function prepareData(method, data) {
+            let payload = null
+            switch(method) {
+                case SessionRequest.methods.personalSign.name: {
+                    payload = SessionRequest.methods.personalSign.getMessageFromData(data)
+                    break
+                }
+                case SessionRequest.methods.sign.name: {
+                    payload = SessionRequest.methods.sign.getMessageFromData(data)
+                    break
+                }
+                case SessionRequest.methods.signTypedData_v4.name: {
+                    const stringPayload = SessionRequest.methods.signTypedData_v4.getMessageFromData(data)
+                    payload = JSON.stringify(JSON.parse(stringPayload), null, 2)
+                    break
+                }
+                case SessionRequest.methods.signTypedData.name: {
+                    const stringPayload = SessionRequest.methods.signTypedData.getMessageFromData(data)
+                    payload = JSON.stringify(JSON.parse(stringPayload), null, 2)
+                    break
+                }
+                default:
```

Comment:
Would it makes sense to case/break SessionRequest.methods sendTransaction and in default case log an error or warning message?